README.md

```diff
@@ -1,247 +1,202 @@
+  - [Understanding Benchmark Runtime](docs/benchmark-runtime.md)
+  - [Configuring kotlinx-benchmark](docs/configuration-options.md)
+  - [Interpreting and Analyzing Results](docs/interpreting-results.md)
+  - [Creating Separate Source Sets](docs/seperate-source-sets.md)
+  - [Tasks Overview](docs/tasks-overview.md)
+  - [Compatibility Guide](docs/compatibility.md)
+  - [Submitting issues and PRs](CONTRIBUTING.md)
 
-Kotlin 1.7.20 or newer
+## Using in Your Projects
 
-# Gradle plugin
+The `kotlinx-benchmark` library is designed to work with Kotlin/JVM, Kotlin/JS, and Kotlin/Native targets. To get started, ensure you're using Kotlin 1.8.20 or newer and Gradle 8.0 or newer.
 
-Use plugin in `build.gradle`:
+### Gradle Setup
 
-```groovy
-plugins {
-    id 'org.jetbrains.kotlinx.benchmark' version '0.4.4'
-}
-```
+<details open>
+<summary>Kotlin DSL</summary>
 
-For Kotlin/JS specify building `nodejs` flavour:
+1.  **Adding Dependency**: Add the `kotlinx-benchmark-runtime` dependency in your `build.gradle.kts` file.
 
-```groovy
-kotlin {
-    js {
-        nodejs()
-        …
-    }   
-}
-```
+    ```kotlin
+    dependencies {
+        implementation("org.jetbrains.kotlinx:kotlinx-benchmark-runtime:0.4.8")
+    }
+    ```
+
+2.  **Applying Benchmark Plugin**: Next, apply the benchmark plugin.
+
+    ```kotlin
+    plugins {
+        id("org.jetbrains.kotlinx.benchmark") version "0.4.8"
+    }
+    ```
+
+3.  **Specifying Repository**: Ensure you have `mavenCentral()` for dependencies lookup in the list of repositories:
+
+        ```kotlin
+        repositories {
+            mavenCentral()
+        }
+        ```
+
+    </details>
+
+<details>
+<summary>Groovy DSL</summary>
+
+1.  **Adding Dependency**: In your `build.gradle` file, include the `kotlinx-benchmark-runtime` dependency.
+
+    ```groovy
+    dependencies {
+        implementation 'org.jetbrains.kotlinx:kotlinx-benchmark-runtime:0.4.8'
+    }
+    ```
 
-For Kotlin/JVM code, add `allopen` plugin to make JMH happy. Alternatively, make all benchmark classes and methods `open`.
+2.  **Applying Benchmark Plugin**: Next, apply the benchmark plugin.
+
+    ```groovy
+    plugins {
+        id 'org.jetbrains.kotlin.plugin.allopen' version "1.8.21"
+        id 'org.jetbrains.kotlinx.benchmark' version '0.4.8'
+    }
+    ```
+    
+3.  **Specifying Repository**: Ensure you have `mavenCentral()` in the list of repositories:
+
+        ```groovy
+        repositories {
+            mavenCentral()
+        }
+        ```
+
+    </details>
+
+### Target-specific configurations
+
+#### Kotlin/JVM
+
+For Kotlin/JVM, applying the [allopen plugin](https://kotlinlang.org/docs/all-open-plugin.html) is pivotal to meet JMH's criteria for `open` benchmark classes/methods. Alternatively, make all benchmark classes and methods `open`. Implement it as follows:
 
-For example, if you annotated each of your benchmark classes with `@State(Scope.Benchmark)`:
```

Comment:
Noted. I have addressed this and added an illustrative example in order to prevent confusion [22e886a](https://github.com/Kotlin/kotlinx-benchmark/pull/117/commits/22e886a9380be053721184ffd87d450175d6a51c)